orca/transform/flagging.py

```diff
@@ -115,3 +118,32 @@ def _calculate_avg_bandpass(autos_corrected, bad_row_mask, n_ants, n_ints, n_cha
     autos_corrected[bad_row_mask] = np.nan
     return bp
 
+def save_flag_metadata(ms: str, output_dir: str = '/lustre/pipeline/slow-averaged/') -> str:
+    """
+    Saves flag metadata in a compact binary format.
+    The output file will have a name derived from the MS name.
+    """
+    base_name = os.path.splitext(os.path.basename(ms))[0]  # e.g., "20241127_220727_73MHz"
+    output_file = os.path.join(output_dir, f"{base_name}_flagmeta.bin")
+
+    tb = table()
+    tb.open(ms)
+    flags = tb.getcol('FLAG')  # (pol, chan, row)
+    tb.close()
+
+    total_points = flags.size
+    flagged_points = np.sum(flags)
+    percentage_flagged = (flagged_points / total_points) * 100.0
+
+    log.info(f"MS: {ms}")
+    log.info(f"Total data points: {total_points}")
+    log.info(f"Flagged points: {flagged_points}")
+    log.info(f"Percentage of flagged data: {percentage_flagged:.2f}%")
+
+    # Pack the flags into a binary format
+    bit_packed = np.packbits(flags.flatten().astype(np.uint8))
```

Comment:
npy takes much more space since it stores each boolean as 8 bits (1 byte) per value. To save space, I decided to use bit packing, which combines 8 booleans into 1 uint8

I am storing the flagmeta.bin files in the same directory as the averaged MS files, like:
/lustre/pipeline/night-time/averaged/73MHz/2023-11-21/03/ 
  └──20231121_035959_73MHz_flagmeta.bin  
  └── 20231121_035959_73MHz_averaged.ms